src/daemon/protocols/cdp.c

```diff
@@ -522,6 +559,21 @@ cdp_decode(struct lldpd *cfg, char *frame, int s,
 			TAILQ_INSERT_TAIL(&port->p_vlans,
 					  vlan, v_entries);
 			break;
+#endif
+#ifdef ENABLE_DOT3
+		case CDP_TLV_POWER_AVAILABLE:
+			CHECK_TLV_SIZE(12, "Power Available");
```

Comment:
You request 12 bytes, but only read 8 bytes. Is that correct?

Reply:
There is a TLV named Power Management Level which seems to be the value the switch wants the PD to display in its Power Consumption TLV. I have not added support for it because it seems unnecessary.
https://www.cisco.com/assets/sol/sb/SG220_Emulators/SG220_Emulator_v1-0-0-18_20140626/help/Discovery17.html